erpnext/controllers/stock_controller.py

```diff
@@ -63,6 +63,12 @@ def validate(self):
 		self.set_rate_of_stock_uom()
 		self.validate_internal_transfer()
 		self.validate_putaway_capacity()
+		self.reset_conversion_factor()
+
+	def reset_conversion_factor(self):
+		for row in self.get("items"):
+			if row.uom and row.conversion_factor != 1.0 and row.uom == row.stock_uom:
```

Comment:
Why message, ideally it should be 1

Reply:
There could be people who are using this as a workaround or so.

![WhatsApp Image 2024-10-15 at 9 23 06 AM](https://github.com/user-attachments/assets/d4812507-be8c-403a-b1a5-77ecda7ea98a)

Reply:
Also, `self.set_rate_of_stock_uom()` is run before this function.
Maybe we should change the order of execution as well?

Reply:
> There could be people who are using this as a workaround or so.
> 
> ![WhatsApp Image 2024-10-15 at 9 23 06 AM](https://private-user-images.githubusercontent.com/52111700/376450948-d4812507-be8c-403a-b1a5-77ecda7ea98a.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5NzM5NDYsIm5iZiI6MTcyODk3MzY0NiwicGF0aCI6Ii81MjExMTcwMC8zNzY0NTA5NDgtZDQ4MTI1MDctYmU4Yy00MDNhLWIxYTUtNzdlY2RhN2VhOThhLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxNVQwNjI3MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xN2M0MjdlOTRmYTQ2N2MzOGU2NmY1MTEyMDQwNTk1ZGNiODZmMDY4ZTBlMGFhNTI3MThiYTM1NzE3ZjNjZTk4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7CXBipdwDp9Iq5pcAzVw84aerdvatzc0J0GvwjDVDOA)

Please share valid example

Reply:
> Please share valid examples

I don't really have a validate use case as such, but since this is an implicit change, informing the user would be nice, just like how we inform when the expense account changes in transactions.

Reply:
Done

Reply:
Thanks!

> Also, `self.set_rate_of_stock_uom()` is run before this function. Maybe we should change the order of execution as well?

Also this ^
Stock Qty and Rate as per Stock UOM fields needs to be set only after this function(?)